run.py

```diff
@@ -74,16 +85,49 @@ def model_state(country, state, interventions=None):
     }
     return CovidTimeseriesModel().forecast(model_parameters=MODEL_PARAMETERS)
 
-if __name__ == '__main__':
-    logging.basicConfig(level=logging.INFO)
-    Dataset = CDSDataset()
-    for state in Dataset.get_all_states_by_country('USA'):
-        for i in range(0, len(INTERVENTIONS)):
-            intervention = INTERVENTIONS[i]
+
+@click.group()
+def main():
+    """Entrypoint for Data Model CLI."""
+    pass
+
+
+@main.command()
+@click.argument('dataset_name', type=click.Choice(['JHU', 'CDS']), default="CDS")
+@click.option(
+    '--output-dir', '-o', default='results/test', help="Model results output directory"
+)
+@click.option('--country', default='USA')
+@click.option('--state', '-s', multiple=True)
+def run_model(dataset_name, output_dir, country, state):
+    # filter_past_date = datetime.date(2020, 3, 19)
+    filter_past_date = None
```

Comment:
Why did you change this to None?  (this is the kind of change I don't want to accidentally sneak in along with refactoring changes, etc. 😛 )

Reply:
heh, in the original implementation, there were multiple calls to create the dataset:
In model_state:
```
Dataset = CDSDataset(filter_past_date=datetime.date(2020, 3, 19))
```
in the main function:
```
Dataset = CDSDataset()
```
This gives one source of truth for the dataset.

What I wanted to do was to add another input, `--filter-past-date` but didn't want to spend time with that parsing logic (also, in python I love using Arrow for datetimes, but that is much more controversial than I wanted to point to)